### ISSUE TITLE
Add support for private repositories to `from_github()`

### ISSUE THREAD

**Issue by gofford** [rstudio/pointblank#290]
## Prework

- [x] Read and abide by **pointblank**'s [code of conduct](https://www.contributor-covenant.org/version/2/0/code_of_conduct/) and [contributing guidelines](https://github.com/rich-iannone/pointblank/blob/master/.github/CONTRIBUTING.md).
- [x] Search for duplicates among the [existing issues](https://github.com/rich-iannone/pointblank/issues) (both open and closed).

## Proposal

`from_github` gives the user a way to retrieve data from github. Example scenarios where this maybe be useful are version controlled schema definitions against which table structure can be compared. However, this function currently only supports **public** repositories. 

I am based on a private organisation and so I cannot use `from_github`. It would be valuable to add support for **private** repositories as well. I am limited of time but am willing to look into this at some point.

**Comment by rich-iannone** [comment]
This sounds like it would be a great addition. Do you also hope to have support for self-hosted GitHub as well?

**Comment by gofford** [comment]
Yeah, I guess it would ideally. Unfortunately I don't have access to an Enterprise hosting at the minute so it'll be hard to develop for. I _do_ have a private org though so I can definitely take a stab at that.

Pretty tight for time for the foreseeable future but this is something I'll like to take a look at once time arises.
